code-challenges.test.js

```diff
@@ -14,39 +20,104 @@
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+describe("capitalizeWords", () => {
+    it("The function should capitalize the first letter of every word in the array", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        expect(capitalizeWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(capitalizeWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+        
+    })
+})
+
+const capitalizeWords = (array) => {
+    let capitalizeArray = []
+    for (let i = 0; i <array.length; i++) {
+        let word = array[i]
+        let capitalizeFirstLetter = word[0].toUpperCase()
+        capitalizeFirstLetter += word.slice(1)
+        capitalizeArray.push(capitalizeFirstLetter)
+    }
+    return capitalizeArray
+}
+
+// console.log(capitalizeWords(randomNouns1))
+// console.log(capitalizeWords(randomNouns2))
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
-// Explain your code:
+// Explain your code: I think I just stayed in my comfort zone and with a for loop. Next time I will take a different approach and .map thought it. I named the function capitalized words, I thought it was appropriate according to the intial directions. But, after looking at the expected output, the result just needed the first letter of each string in the array to be capatilzed. I put capitlized Array as an empty array. I loop through the arrays using a for loop. I created a variable called word, and it targets each individual string in the array. I also created a variable named capitalizeFirstLetter which targets and uppercases the first letter in the "word" variable in the array.  Then this line  (capitalizeFirstLetter += word.slice(1)) basically puts the first capitalized letter back together with the rest of the sting. The .slice had to start with the second letter and concatenate with the first capitalized letter. Then I used .push to  push it into the capitalizeArray variable and return it.
```

Comment:
great job and I love you did what you know.  Go back and refactor it when you have a chance.  You can do it!